### ISSUE TITLE
KeyError: 'anchors' when training with --evolve

### ISSUE THREAD

**Issue by sinhau** [ultralytics/yolov5#2485]
## 🐛 Bug
When I run training with the --evolve parameter, I get a `KeyError: 'anchors'` error.

Digging into the train.py script, it looks like the `meta` variable defined in line 536 contains the `anchor` param but that param is commented out in the default hyper parameter file located at  `data/hyp.scratch.yaml`.  Is this intentional or a bug?

## To Reproduce (REQUIRED)

Input:
```
python train.py --epochs 10 --evolve
```

Output:
```
github: up to date with https://github.com/ultralytics/yolov5 ✅
YOLOv5 🚀 v4.0-138-ged2c742 torch 1.7.1 CUDA:0 (Tesla V100-SXM2-16GB, 16160.5MB)
Namespace(adam=False, batch_size=16, bucket='', cache_images=False, cfg='', data='data/coco128.yaml', device='', entity=None, epochs=10, evolve=True, exist_ok=False, global_rank=-1, hyp='data/hyp.scratch.yaml', image_weights=False, img_size=[640,
 640], linear_lr=False, local_rank=-1, log_artifacts=False, log_imgs=16, multi_scale=False, name='evolve', noautoanchor=False, nosave=False, notest=False, project='runs/train', quad=False, rect=False, resume=False, save_dir='runs/train/evolve', s
ingle_cls=False, sync_bn=False, total_batch_size=16, weights='yolov5s.pt', workers=8, world_size=1)
Traceback (most recent call last):
  File "train.py", line 600, in <module>
    hyp[k] = max(hyp[k], v[1])  # lower limit
KeyError: 'anchors'
```

## Environment

 - OS: Debian 10
 - GPU: Tesla V100

**Comment by glenn-jocher** [comment]
@sinhau yes you are right! hyp anchors will override yaml anchors if present (uncommented), but for hyp evolution this line should be uncommented, and then your anchor anchor count will evolve along with the other hyps.

I know this workflow isn't the best, I'm not quite sure how to fix this, but I'm open to ideas.
https://github.com/ultralytics/yolov5/blob/ed2c74218d6d46605cc5fa68ce9bd6ece213abe4/data/hyp.scratch.yaml#L20

**Comment by glenn-jocher** [comment]
@sinhau good news 😃! Your original issue may now be fixed ✅ in PR #3887. This PR adds `hyp['anchors']=3` if the anchors key is missing/commented in your hyperparameter file, fixing a long-running problem when starting hyp evolution using the default command:
```
python train.py --evolve
```

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
